lib/utils/normalize-manufacturer-part-number.ts

```diff
@@ -1,13 +1,10 @@
 export function normalizeManufacturerPartNumber(partNumber: string): string {
-  // Step 1: Replace dashes with underscores
-  let normalized = partNumber.replace(/-/g, "_")
+  // Step 1: Replace dashes and invalid symbols with underscores
+  let normalized = partNumber.replace(/[-]/g, "_").replace(/[^a-zA-Z0-9_$]/g, "_")
 
-  // Step 2: Remove all invalid symbols
-  normalized = normalized.replace(/[^a-zA-Z0-9_$]/g, "")
-
-  // Step 3: If the string starts with a number, prepend 'A'
+  // Step 2: If the string starts with a number, prepend 'A_'
   if (/^\d/.test(normalized)) {
-    normalized = "A" + normalized
+    normalized = "A_" + normalized
```

Comment:
Decided to prepend `A_` instead of just `A` to make it more clear that it's not part of manufacturer part number